### PR TITLE
artifactory (non-ha) master-key support and ingress default backend toggle support

### PR DESCRIPTION
**Pull request by zendril**
@jainishshah17 , @eldada , @rimusz 

**What this PR does / why we need it**:

- The non-ha version of artifactory does not start up without a master-key. This adds the ability to specify master key in the values. It defaults to the same key that is used for the artifactory-ha version: FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
- Also, on some clusters, the ability to run the default `backend` ingress is disabled. This PR adds the ability to toggle it on off via the values. It defaults to the current behavior of enabled.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
Discussed some of these challenges with David Xu (our support rep from JFrog). Please feel free to reach out thru him to setup a call with me to discuss if that helps.

### CODE REVIEW

File: stable/artifactory/templates/ingress.yaml [hunk 1]
```diff
@@ -15,9 +15,12 @@ metadata:
       {{ $key }}: {{ $value | quote }}
     {{- end }}
 spec:
+  {{- if .Values.ingress.defaultBackend.enabled }}
```
Comment (eldada) [inline stable/artifactory/templates/ingress.yaml]: Please add this to `artifactory-ha` as well. We like to keep them aligned where possible.

File: stable/artifactory/values.yaml [hunk 2]
```diff
@@ -39,6 +39,8 @@ serviceAccount:
 
 ingress:
   enabled: false
+  defaultBackend:
```
Comment (eldada) [inline stable/artifactory/values.yaml]: Please add this to `artifactory-ha` as well. We like to keep them aligned where possible.

File: stable/artifactory/templates/ingress.yaml (hunk 1 above)
Comment (zendril) [reply stable/artifactory/templates/ingress.yaml]: Can do, Altho I don't have artifactory-ha running on my cluster right now so I cannot fully test it.

Also, sorry about the README.md, I had already modified it but left the file out of the commit. :/

Comment (zendril) [reply stable/artifactory/templates/ingress.yaml]: done in 9e2a15175f76af2e5baba908b28f9d07329e4ffa

File: stable/artifactory/values.yaml (hunk 2 above)
Comment (zendril) [reply stable/artifactory/values.yaml]: done in 9e2a15175f76af2e5baba908b28f9d07329e4ffa

**Comment by zendril** [comment]
I added the same defaultBackend support into xray and I also made it easier to swap out rabbitmq-ha for rabbitmq (altho I didn't explicitly pull in rabbitmq into the requirements).

**Comment by zendril** [comment]
hang tight before reviewing again, i'm going to go ahead and add in the ability to pull in rabbitmq so the user doesn't have to jump thru extra hoops.

**Comment by zendril** [comment]
ok.. go ahead and review please (once travis gets happy)

**Comment by jainishshah17** [comment]
@zendril Yes its known issue, We are working on it.

**Comment by jainishshah17** [comment]
@zendril Can you please fix following:
```
stable/xray/values.yaml
  32:18     error    trailing spaces  (trailing-spaces)
  168:17    error    trailing spaces  (trailing-spaces)
```

**Comment by zendril** [comment]
@jainishshah17 done.

**Comment by jainishshah17** [comment]
@zendril Awesome! Now it's passing linting. I will merge after a small manual test.

**Pull request merged**